### DISCUSSION TITLE
얕은 복사와 깊은 복사에 대해 설명해주세요.

### DISCUSSION THREAD

**Discussion by pipisebastian** [publdaze/js-study#47] [원시 값과 객체의 비교]
- 얕은 복사와 깊은 복사에 대해 설명해주세요.
  - 얕은 복사 방법에는 무엇이 있나요?
  - 깊은 복사 방법에는 무엇이 있나요?

**Comment by pipisebastian** [comment]
> 얕은 복사와 깊은 복사에 대해 설명해주세요.

- `얕은 복사` : 한단계만. 참조값을 복사
- `깊은 복사` : 중첩객체까지 싹다.
    
> 얕은 복사 방법에는 무엇이 있나요?

- =, Object.assign(), 스프레드 연산자

> 깊은 복사 방법에는 무엇이 있나요?
    
- JSON.parse && JSON.stringify, lodash

**Comment by publdaze** [comment]
> - 얕은 복사와 깊은 복사에 대해 설명해주세요.
- 얕은 복사는 객체의 최상위 수준만 복사하는 것으로 변수에 저장된 값인 참조값이 복사되어 복사된 값을 가지는 변수 또한 동일한 참조값을 통해 같은 데이터를 바라봄
- 깊은 복사는 객체의 모든 수준을 복사하는 것으로 하위 프로퍼티 값또한 모두 복사가 된 것을 의미
> - 얕은 복사 방법에는 무엇이 있나요?
- 할당 연산자를 활용하여 객체 변수를 다른 변수에 할당하는 방법, slice 메소드, 스프레드 연산자, Object.assign 메소드 사용하는 방법
> - 깊은 복사 방법에는 무엇이 있나요?
- lodash 라이브러리의 cloneDeep 메서드 사용하는 방법, JSON.stringify 메소드를 통해 json 형태로 변환한 후 JSON.parse 메소드로 다시 파싱을 하는 방법
